server/issue_test.go

```diff
@@ -166,3 +166,55 @@ func TestCheckCharacterSetCollationOwner(t *testing.T) {
 		}
 	}
 }
+
+func TestGetTaskAndDagListByVersion(t *testing.T) {
+	databaseID1 := 1
+	databaseID2 := 2
+	tests := []struct {
+		versionTaskList []*versionTask
+		wantTaskList    []api.TaskCreate
+		wantDAGList     []api.TaskIndexDAG
+	}{
+		{
+			versionTaskList: []*versionTask{},
+			wantTaskList:    nil,
+			wantDAGList:     nil,
+		},
+		{
+			versionTaskList: []*versionTask{
+				{
+					task:    &api.TaskCreate{DatabaseID: &databaseID1, Name: "task2"},
+					version: "v2",
+				},
+				{
+					task:    &api.TaskCreate{DatabaseID: &databaseID2, Name: "task2"},
+					version: "v2",
+				},
+				{
+					task:    &api.TaskCreate{DatabaseID: &databaseID1, Name: "task3"},
+					version: "v3",
+				},
+				{
+					task:    &api.TaskCreate{DatabaseID: &databaseID1, Name: "task1"},
+					version: "v1",
+				},
+			},
+			wantTaskList: []api.TaskCreate{
+				{DatabaseID: &databaseID1, Name: "task1"},
+				{DatabaseID: &databaseID1, Name: "task2"},
+				{DatabaseID: &databaseID1, Name: "task3"},
+				{DatabaseID: &databaseID2, Name: "task2"},
+			},
+			wantDAGList: []api.TaskIndexDAG{
+				{FromIndex: 0, ToIndex: 1},
+				{FromIndex: 1, ToIndex: 2},
```

Comment:
How can we map the DAG to a specific database?

Seems to be hard to figure out without an explicit database ID here.

Reply:
The FromIndex/ToIndex are the indices of tasks in the task list. It's possible to have DAG across databases.